command: update edit-validation-sets to craft-cli

sergiusens opened the pull request:
This is a minimal code change migration from Snapcraft's snapcraft_legacy module to snapcraft. Adaptations to use craft-cli for output has been made, as well as updates to how file edit takes place by migrating it to use Path.

The tests have been updated to follow the newer conventions of the code base, but the gist of them has not changed at all.

Lastly, the only UI change is that validation sets have been separated into their own command group.

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `make lint`?
- [x] Have you successfully run `pytest tests/unit`?

-----
CRAFT-1402

tigarmo commented:
It's a bit hard to tell what's code that has just been moved and what's code that's materially changed. Can you point at which changes we should pay special attention to?

sergiusens commented on snapcraft/cli.py:
```diff
@@ -101,18 +101,23 @@
         ],
     ),
+    craft_cli.CommandGroup(
+        "Store Validation Sets",
```
I moved the commands to a new command group

sergiusens commented on snapcraft/commands/validation_sets.py:
```diff
@@ -0,0 +1,166 @@
+            edited_validation_sets = _edit_validation_sets(unverified_validation_sets)
+
+        if edited_validation_sets == yaml.safe_load(unverified_validation_sets):
+            emit.message("No changes made")
```
click's echo was replaced with emit

sergiusens commented on snapcraft/commands/validation_sets.py:
```diff
@@ -0,0 +1,166 @@
+    editor_cmd = os.getenv("EDITOR", "vi")
+
+    with tempfile.NamedTemporaryFile() as temp_file:
+        file_template_path = Path(temp_file.name)
```
the context management here has been simplified thanks to use of Path and its write_text and read_text methods

sergiusens commented:
@tigarmo I added the relevant comments, the changes are really simple in themselves and out of necessity as "click" needs to be replaced with emit; and pydantic required changes in the edit method so I moved it to Path which simplified it

tigarmo commented:
thanks, that helped a lot!
